From bd0a57886f68ecd08b87cccb942c1b0634c19509 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Sat, 23 Jun 2018 13:18:27 +0200
Subject: [PATCH 22/24] SystemUI: CellularTile: Don't call showDetail() when
 device is locked

* We recently enabled dualTarget for CellularTile which allowed us
  to toggle mobile data without unlocking device. Moving showDetail
  call to postQSRunnableDismissingKeyguard when the device is locked
  fixes this issue.

Change-Id: I16dc766166a33a309a2fc945e81fa15b5d194196
---
 .../src/com/android/systemui/qs/tiles/CellularTile.java     | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/CellularTile.java b/packages/SystemUI/src/com/android/systemui/qs/tiles/CellularTile.java
index c0b0473b1e7..5014b0cfea3 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/CellularTile.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/CellularTile.java
@@ -142,6 +142,12 @@ public class CellularTile extends QSTileImpl<SignalState> {
     @Override
     protected void handleSecondaryClick() {
         if (mDataController.isMobileDataSupported()) {
+            if (mKeyguardMonitor.isSecure() && !mKeyguardMonitor.canSkipBouncer()) {
+                mActivityStarter.postQSRunnableDismissingKeyguard(() -> {
+                    showDetail(true);
+                });
+                return;
+            }
             showDetail(true);
         } else {
             mActivityStarter
-- 
2.17.1

